notes.md

```diff
@@ -0,0 +1,215 @@
+# Why does `:` only sometimes work?
+I don't like `:` being listed as a `source_column_selector` when it often errors.
+
+`select(df, :)`
+: works
+
+`transform(df, Cols(:) => ByRow(max))`
+: works
+
+`transform(df, : => ByRow(max))` errors
+: ERROR: syntax: whitespace not allowed after ":" used for quoting
+
+Similarly, regular expressions also sometimes must be wrapped in `Cols`.
+This seems inconsistent.
+I don't understand the rules for when `Cols` is and isn't needed.
+It should be explained in the documentation.
+
+```julia
+julia> df = DataFrame(Time=[3, 4, 5], TopTemp=[70, 73, 100], BottomTemp=[50, 55, 80])
+3×3 DataFrame
+ Row │ Time   TopTemp  BottomTemp
+     │ Int64  Int64    Int64
+─────┼────────────────────────────
+   1 │     3       70          50
+   2 │     4       73          55
+   3 │     5      100          80
+
+julia> transform(df, Cols(r"Temp") .=> (t->(t.-32)*5/9), renamecols=false)
+3×3 DataFrame
+ Row │ Time   TopTemp  BottomTemp
+     │ Int64  Float64  Float64
+─────┼────────────────────────────
+   1 │     3  21.1111     10.0
+   2 │     4  22.7778     12.7778
+   3 │     5  37.7778     26.6667
+
+julia> transform(df, r"Temp" .=> (t->(t.-32)*5/9), renamecols=false) # breaks without Cols
+ERROR: MethodError: no method matching (::var"#5#6")(::Vector{Int64}, ::Vector{Int64})
+```
+```julia
+julia> df = DataFrame([1 2 3 4], :auto)
+1×4 DataFrame
+ Row │ x1     x2     x3     x4
+     │ Int64  Int64  Int64  Int64
+─────┼────────────────────────────
+   1 │     1      2      3      4
+
+julia> select(df, r"x" => ByRow(min)) # works without Cols
+1×1 DataFrame
+ Row │ x1_x2_etc_min
+     │ Int64
+─────┼───────────────
+   1 │             1
+```
+
+# Cannot pack columns into a named tuple column?
+If I can do this to unpack named tuples:
+```julia
+julia> df = DataFrame(in=[(a=1,b=2), (a=3, b=4)])
+2×1 DataFrame
+ Row │ in
+     │ NamedTup…
+─────┼────────────────
+   1 │ (a = 1, b = 2)
+   2 │ (a = 3, b = 4)
+
+julia> df2 = select(df, :in => AsTable)
+2×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      2
+   2 │     3      4
+```
+Then I should also be able to do the reverse to pack named tuples and get `df` back:
+```julia
+julia> df3 = select(df2, AsTable(:) => :in)
+ERROR: ArgumentError: Unrecognized column selector: AsTable(Colon()) => :in
+```
+
+# Why are types truncated when there is still a lot of room?
+
+Also, why is `NamedTup...` truncated in the output above? There is room.
```

Comment:
because when typesetting the header we do not know the column width yet. However, maybe @ronisbr could improve this in the future.

Reply:
Hi everyone!

When we were discussing how to use PrettyTables.jl to print DataFrames, we decided to use the previous mechanism to provide the type name (the function is called `batch_compacttype`). We can avoid those "unnecessary" contractions by using `crop_subheader = true` and specifying a minimum size for the columns. For example:

```
julia> df = DataFrame(a = (a = 1, b = 2), b = (c = 3, d = 4));

julia> show(df, header = (names(df), [eltype(c) for c in eachcol(df)]), crop_subheader = true)
1×2 DataFrame
 Row │ a               b
     │ NamedTuple{(:…  NamedTuple{(:…
─────┼────────────────────────────────
   1 │ (a = 1, b = 2)  (c = 3, d = 4)
```

Notice that the type sometimes is strange due to the cropping. That was the reason we decided to keep the previous behavior.

Reply:
Yes! Sure! I will add to the list :)